flyteidl/protos/flyteidl/core/metrics.proto

```diff
@@ -34,3 +35,32 @@ message Span {
     // spans defines a collection of Spans that breakdown this execution.
     repeated Span spans = 7;
 }
+
+enum ExecutionMetric {
```

Comment:
It will be challenging to anticipate or keep up with all possible metrics (and aggregations). I suspect at minimum we will want some string escape hatch, i.e., `oneof {enum string}`, if not just a string encoding the query (metric + aggregation)